cpp/include/cudf/io/types.hpp

```diff
@@ -277,6 +277,8 @@ struct column_name_info {
 struct table_metadata {
   std::vector<column_name_info>
     schema_info;  //!< Detailed name information for the entire output hierarchy
+  std::vector<size_t> num_rows_per_source;  //!< Number of rows read from each data source. Empty
+                                            //!< vector if AST filters are being used.
```

Comment:
hm, this is shared between all readers. The comment should reflect that only Parquet supports this